### PR TITLE
[CI] Add option to raise an exception when NaNs are detected in logits

### PR DESCRIPTION
**Pull request by tlrmchlsmth**
This PR sets `VLLM_COMPUTE_NANS_IN_LOGITS=1` in several evals so we can test against NaNs. 

NaNs appearing in logits often co-appear with KV cache NaNs, which is a catastrophic failure mode for a inference service since attention kernels mask with multiplication by zero. (see https://github.com/Dao-AILab/flash-attention/issues/1974)

Assisted-by: OpenAI Codex

### CODE REVIEW

**Comment by tlrmchlsmth** [comment]
@claude review

File: .buildkite/scripts/scheduled_integration_test/deepseek_v2_lite_ep_eplb.sh [hunk 1]
```diff
@@ -42,6 +42,7 @@ cleanup() {
 trap cleanup EXIT
 
 for BACK in "${BACKENDS[@]}"; do
+  VLLM_COMPUTE_NANS_IN_LOGITS=1 \
```
Comment (AndreasKaratzas) [inline .buildkite/scripts/scheduled_integration_test/deepseek_v2_lite_ep_eplb.sh]: should we let each hw backend determine the value of that? tbh if this goes through i would ask if you could set it in `run-am-tests.sh` so that it is enabled by default on amd for everything. or you think that it should be kept in test level?

**Comment by njhill** [comment]
Thanks @tlrmchlsmth this looks great!

I wonder though, rather than relying on scraping the metrics, we could maybe have a `VLLM_COMPUTE_NANS_IN_LOGITS` mode where an exception is raised if a NaN is encountered. Then we can just set that globally for the whole CI (doesn't even need to be added per test). Basically it would just blow up the test if encountered.

**Comment by mgoin** [comment]
^ +1 to Nick's suggestion! I'd like this to be global on CI and raise so the pytest/script fails

**Comment by tlrmchlsmth** [comment]
Going to add `VLLM_RAISE_ON_LOGIT_NANS` or something and then set it in all of buildkite

**Comment by tlrmchlsmth** [comment]
partner PR is up here https://github.com/vllm-project/ci-infra/pull/444

**Review by njhill — APPROVED** [review]
Thanks @tlrmchlsmth LGTM

**Comment by tlrmchlsmth** [comment]
Switching the envs to 1 to see if anything breaks -- so nobody merge it yet ;)

**Review by NickLucche — APPROVED** [review]
